src/ifk_scb_compilations/objects/passenger_transport.py

```diff
@@ -0,0 +1,127 @@
+            },
+            {
+                "code": "Sektor",
+                "selection": {"filter": "item", "values": ["0.2", "0.4", "8.0", "5.0"]},
+            },
+        ],
+        "response": {"format": "json"},
+    }
+
+
+class Analysis:
+    """Container for plotting data corresponding to fetch spec by Request_input."""
+
+    def __init__(self, request_output: dict) -> None:
+        """Initialization.
+
+        Args:
+            request_output (dict): Output from scb api response.
+        """
+        self.data = self.transform_json_to_df(request_output)
+
+    def transform_json_to_df(self, request_output: dict) -> pd.DataFrame:
+        """Transform json format to dataframe.
+
+        Args:
+            request_output (dict): The first parameter.
+
+        Returns:
+            pd.Dataframe: request output as dataframe
+        """
+        n_vals = len(request_output["data"])
+
+        data_dict = {
+            "emission measure": list(
+                request_output["data"][i]["key"][0] for i in range(n_vals)
+            ),
+            "emission type": list(
+                request_output["data"][i]["key"][1] for i in range(n_vals)
+            ),
+            "year": list(
+                float(request_output["data"][i]["key"][2]) for i in range(n_vals)
+            ),
+            "value": list(
+                float(request_output["data"][i]["values"][0]) for i in range(n_vals)
+            ),
+        }
+
+        return pd.DataFrame.from_dict(data_dict)
+
+    def plot_co2_transports(self) -> None:
+        """Plot CO2 for transports."""
+        labels = {
+            "0.2": "NATIONELL TOTAL (exklusive LULUCF, inklusive internationella transporter)",
+            "0.4": "NATIONELL TOTAL (inklusive LULUCF, inklusive internationella transporter)",
+            "8.0": "INRIKES TRANSPORTER, TOTALT",
+            "5.0": "UTRIKES TRANSPORTER, TOTALT",
+        }
+
+        def _plot_individual(emission_type: str) -> None:
+            """Plot based on emission type.
+
+            Args:
+                emission_type (str): emission type code
+            """
+            ax.plot(
+                self.data[self.data["emission type"] == emission_type]["year"],
+                self.data[self.data["emission type"] == emission_type]["value"],
+                label=labels[emission_type],
+            )
+
+        ax = plt.subplot(111)
+        _plot_individual("0.2")
+        _plot_individual("0.4")
+        _plot_individual("8.0")
+        _plot_individual("5.0")
+
+        ax.set_title("Utsläpp av växthusgaser i CO2-ekvivalent. Källa:SCB")
+        ax.set_xlabel("År")
+        ax.set_ylabel("kiloTon CO2-ekvivalent")
+        ax.legend(loc="upper center", bbox_to_anchor=(0.5, -0.05))
+        box = ax.get_position()
+        ax.set_position(
+            [box.x0, box.y0 + box.height * 0.1, box.width, box.height * 0.9]
+        )
+        plt.show()
+        pass
```

Comment:
Is this needed, and in the method below?

Reply:
removed.